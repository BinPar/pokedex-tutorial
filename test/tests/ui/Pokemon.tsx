import React from 'react';
import { mount } from 'enzyme';
import mockNextUseRouter from '../../mocks/mockNextUseRouter';
import Pokemon, {getStaticProps} from '../../../pages/pokemon/[name]';


/**
 * Checks that the Home Page redirects to some sample
 */
test('Pokemon Page', async (): Promise<void> => {
  /**
   * Mocks the next router
   */
  mockNextUseRouter({
    route: '/pokemon/bulbasaur',
    pathname: '/pokemon/bulbasaur',
    query: {
      name: 'bulbasaur'
    },
    asPath: '/pokemon/bulbasaur',
  });

  const serverSideProps = await getStaticProps({
    params: {name: 'bulbasaur'},
  });
  // Renders the page
  const page = mount(<Pokemon {...serverSideProps.props}  />);

  // We check that the content of the page is not changed
  expect(page).toMatchSnapshot('expanded');

  expect(page.find('h1')).toHaveLength(1);
  expect(page.find('h1').first().text()).toBe('#1 bulbasaur');
  expect(page.find('h2 a')).toHaveLength(2);
  page.find('h2 a').forEach((item) => {
  expect(item.text()[0]).toBe('-');
  item.simulate('click');
  expect(item.text()[0]).toBe('+');
  });

  expect(page).toMatchSnapshot('non expanded');

});
