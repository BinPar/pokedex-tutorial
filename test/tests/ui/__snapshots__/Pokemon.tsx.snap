// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon Page: expanded 1`] = `
<PokemonPage
  pokemon={
    Object {
      "height": 7,
      "id": 1,
      "imageURL": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png",
      "name": "bulbasaur",
      "stats": Object {
        "attack": 49,
        "defense": 49,
        "hp": 45,
        "special-attack": 65,
        "special-defense": 65,
        "speed": 45,
      },
      "types": Array [
        "grass",
        "poison",
      ],
      "weight": 69,
    }
  }
>
  <Head>
    <_default
      headManager={Object {}}
      inAmpMode={false}
      reduceComponentsToState={[Function]}
    />
  </Head>
  <main>
    <h1>
      #1 bulbasaur
    </h1>
    <img
      alt="bulbasaur"
      src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
    />
    <h2>
      <a
        href=""
        onClick={[Function]}
      >
        <strong
          style={
            Object {
              "display": "block",
              "float": "left",
              "width": 40,
            }
          }
        >
          -
        </strong>
        Info
      </a>
    </h2>
    <dl>
      <dt>
        Height:
      </dt>
      <dd>
        7
      </dd>
      <dt>
        Weight:
      </dt>
      <dd>
        69
      </dd>
      <dt>
        Types:
      </dt>
      <dd>
         grass poison
      </dd>
    </dl>
    <h2>
      <a
        href=""
        onClick={[Function]}
      >
        <strong
          style={
            Object {
              "display": "block",
              "float": "left",
              "width": 40,
            }
          }
        >
          -
        </strong>
        Stats
      </a>
    </h2>
    <dl>
      <dt>
        hp: 
      </dt>
      <dd>
        45
      </dd>
      <dt>
        attack: 
      </dt>
      <dd>
        49
      </dd>
      <dt>
        defense: 
      </dt>
      <dd>
        49
      </dd>
      <dt>
        special-attack: 
      </dt>
      <dd>
        65
      </dd>
      <dt>
        special-defense: 
      </dt>
      <dd>
        65
      </dd>
      <dt>
        speed: 
      </dt>
      <dd>
        45
      </dd>
    </dl>
  </main>
  <div
    style={
      Object {
        "clear": "both",
      }
    }
  >
    <Link
      href="/"
    >
      <a
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Ir a home
      </a>
    </Link>
  </div>
</PokemonPage>
`;

exports[`Pokemon Page: non expanded 1`] = `
<PokemonPage
  pokemon={
    Object {
      "height": 7,
      "id": 1,
      "imageURL": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png",
      "name": "bulbasaur",
      "stats": Object {
        "attack": 49,
        "defense": 49,
        "hp": 45,
        "special-attack": 65,
        "special-defense": 65,
        "speed": 45,
      },
      "types": Array [
        "grass",
        "poison",
      ],
      "weight": 69,
    }
  }
>
  <Head>
    <_default
      headManager={Object {}}
      inAmpMode={false}
      reduceComponentsToState={[Function]}
    />
  </Head>
  <main>
    <h1>
      #1 bulbasaur
    </h1>
    <img
      alt="bulbasaur"
      src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
    />
    <h2>
      <a
        href=""
        onClick={[Function]}
      >
        <strong
          style={
            Object {
              "display": "block",
              "float": "left",
              "width": 40,
            }
          }
        >
          +
        </strong>
        Info
      </a>
    </h2>
    <dl />
    <h2>
      <a
        href=""
        onClick={[Function]}
      >
        <strong
          style={
            Object {
              "display": "block",
              "float": "left",
              "width": 40,
            }
          }
        >
          +
        </strong>
        Stats
      </a>
    </h2>
    <dl />
  </main>
  <div
    style={
      Object {
        "clear": "both",
      }
    }
  >
    <Link
      href="/"
    >
      <a
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Ir a home
      </a>
    </Link>
  </div>
</PokemonPage>
`;
